### DISCUSSION TITLE
QR code

### DISCUSSION THREAD

**Discussion by DBR-it** [FireMarshmellow/Spotlight_Storage#52] [Ideas]
Maybe a future item!
QR code for each individual item. When scanned, the QR code would link to that specific item, allowing users to easily update the quantity they've taken.

**Comment by GitHag42** [comment]
QR codes would be a super nice feature!

Having seen [this](https://github.com/SmartHome-yourself/barcode-scanner-for-esphome) project with a barcode/qr code scanner connected to an ESP32 I have the following two ideas in mind.

Requirements:
- each bin has a QR code on the front label or on the side
- there is a scanner connected to the ESP32 of the spotlight storage or to an additional ESP32

When sitting at the desk with the storage:
- with multiple bins taken out one could scan a bin and the according place lights up and it is easy to put it back in the correct location
- with the potential to export the QR codes from the database/web frontend or (better) to create a BOM with QR codes included one could scan the parts from the printed BOM and find the according bins easily

Both ideas together would allow to use the system without having to use the web frontend. So one could prepare things (BOM or bins on the table) for someone and he/she can find bins or put bins back where they belong without needing to use the web search.

Thoughts:
- The QR codes should be generated from the web frontend and there need to be a way to export them as graphic (for a BOM) and the data only to print the codes on a label printer for instance. Most label printers can generate QR codes themselves.
- The QR codes should be linked to the parts as an attribute (to simulate a search for the part name) and not to the location. This way the bin including the part, a label and the QR code could be relocated if things should be organized in a new way (due to new/more parts).
- I mainly focused on finding the location, but one could think of using additional qr codes for some actions, such as adding items to a bin (refill) or substracting items from a bin. Scan the bin, scan a code for removing items, scan a code for a number...
- (The display would also be nice to show the according part name when scanning the QR code. But that is not really necessary. The bins are probably labeled.)
